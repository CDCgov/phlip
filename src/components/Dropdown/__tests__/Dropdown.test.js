import React from 'react'
import { shallow } from 'enzyme'
import { Dropdown } from '../index'

const props = {
  shrinkLabel: true,
  required: false,
  options: [{ label: 'Option 1', value: 1 }, { label: 'Option 2', value: 2 }],
  meta: { touched: false, error: undefined },
  label: '',
  displayEmpty: false,
  fullWidth: false,
  input: { value: 1 },
  id: 'test-dropdown',
  classes: {
    disabled: 'class-disabled',
    icon: 'class-icon',
    disabledIcon: 'class-disabled-icon',
    disabledLabel: 'class-disabled-label'
  }
}

describe('Dropdown', () => {
  test('should render correctly', () => {
    expect(shallow(<Dropdown {...props} />)).toMatchSnapshot()
  })
})
