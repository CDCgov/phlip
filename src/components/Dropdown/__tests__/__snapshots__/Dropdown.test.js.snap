// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render correctly 1`] = `
<FlexGrid
  align="flex-start"
  circular={false}
  container={true}
  flex={false}
  id="test-dropdown-container"
  justify="stretch"
  padding={0}
  raised={false}
  style={
    Object {
      "minWidth": 120,
    }
  }
  type="column"
>
  <WithStyles(Select)
    classes={
      Object {
        "disabled": "class-disabled",
        "icon": "class-icon",
      }
    }
    disabled={false}
    displayEmpty={false}
    input={
      <WithStyles(Input)
        id="test-dropdown"
      />
    }
    meta={
      Object {
        "error": undefined,
        "touched": false,
      }
    }
    onChange={[Function]}
    value={1}
  >
    <WithStyles(MenuItem)
      key="menu-item-0"
      value={1}
    >
      Option 1
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="menu-item-1"
      value={2}
    >
      Option 2
    </WithStyles(MenuItem)>
  </WithStyles(Select)>
</FlexGrid>
`;
